rpc-client-api/src/response.rs

```diff
@@ -377,13 +378,13 @@ pub struct RpcIdentity {
     pub identity: String,
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+#[derive(Serialize, Deserialize, Clone, Debug)]
 #[serde(rename_all = "camelCase")]
 pub enum RpcBundleSimulationSummary {
-    /// error and offending transaction signature
+    /// error and offending transaction signature if applicable
     Failed {
-        error: (),
-        tx_signature: String,
+        error: BundleExecutionError,
```

Comment:
might make sense to move our custom return type into this file + map what v1.14 had (maybe with some etxra stuff?)